lib/src/main/java/graphql/nadel/instrumentation/parameters/NadelInstrumentationTimingParameters.kt

```diff
@@ -0,0 +1,88 @@
+package graphql.nadel.instrumentation.parameters
+
+import graphql.execution.instrumentation.InstrumentationState
+import graphql.nadel.engine.transform.NadelTransform
+import java.time.Duration
+import kotlin.reflect.KClass
+
+data class NadelInstrumentationTimingParameters(
+    val step: Step,
+    val duration: Duration,
+    /**
+     * If an exception occurred during the timing of the step, then it is passed in here.
+     */
+    val exception: Throwable?,
+    private val context: Any?,
+    private val instrumentationState: InstrumentationState?,
+) {
+    fun <T> getContext(): T? {
+        @Suppress("UNCHECKED_CAST") // trust the caller
+        return context as T
+    }
+
+    fun <T : InstrumentationState?> getInstrumentationState(): T? {
+        @Suppress("UNCHECKED_CAST") // trust the caller
+        return instrumentationState as T?
+    }
+
+    sealed interface Step {
```

Comment:
Ok maybe slightly overkill but externally it's a simple interface that just has `parent` and `name` of the step.

Then you can use `getFullName()` to get a name for the metric e.g. `QueryTransforming.HydrationTransform`